yarn-project/acir-simulator/src/avm/opcodes/arithmetic.ts

```diff
@@ -0,0 +1,154 @@
+
+    // TODO: proper field division
+    const dest = new Fr(a.toBigInt() / b.toBigInt());
+    context.writeMemory(this.destOffset, dest);
+  }
+}
+
+/** -*/
+export class Eq implements Opcode {
+  static type: string = 'EQ';
+  static numberOfOperands = 3;
+
+  constructor(private aOffset: number, private bOffset: number, private destOffset: number) {}
+
+  execute(context: AvmContext, _stateManager: AvmStateManager): void {
+    const a: Fr = context.readMemory(this.aOffset);
+    const b: Fr = context.readMemory(this.bOffset);
+
+    // TODO: floor div? - this will not perform field division
+    const dest = new Fr(a.toBigInt() == b.toBigInt());
+    context.writeMemory(this.destOffset, dest);
+  }
+}
+/** -*/
+export class Lt implements Opcode {
+  static type: string = 'Lt';
+  static numberOfOperands = 3;
+
+  constructor(private aOffset: number, private bOffset: number, private destOffset: number) {}
+
+  execute(context: AvmContext, _stateManager: AvmStateManager): void {
+    const a: Fr = context.readMemory(this.aOffset);
+    const b: Fr = context.readMemory(this.bOffset);
+
+    // TODO: floor div? - this will not perform field division
+    const dest = new Fr(a.toBigInt() < b.toBigInt());
+    context.writeMemory(this.destOffset, dest);
+  }
+}
+
+/** -*/
+export class Lte implements Opcode {
+  static type: string = 'LTE';
+  static numberOfOperands = 3;
+
+  constructor(private aOffset: number, private bOffset: number, private destOffset: number) {}
+
+  execute(context: AvmContext, _stateManager: AvmStateManager): void {
+    const a: Fr = context.readMemory(this.aOffset);
+    const b: Fr = context.readMemory(this.bOffset);
+
+    // TODO: floor div? - this will not perform field division
+    const dest = new Fr(a.toBigInt() < b.toBigInt());
+    context.writeMemory(this.destOffset, dest);
+  }
+}
+
+/** -*/
+export class Shl implements Opcode {
+  static type: string = 'SHL';
+  static numberOfOperands = 3;
+
+  constructor(private aOffset: number, private bOffset: number, private destOffset: number) {}
+
+  execute(context: AvmContext, _stateManager: AvmStateManager): void {
+    const a: Fr = context.readMemory(this.aOffset);
+    const b: Fr = context.readMemory(this.bOffset);
+
+    // TODO: floor div? - this will not perform field division
+    const dest = new Fr(a.toBigInt() << b.toBigInt());
+    context.writeMemory(this.destOffset, dest);
+  }
+}
+
+/** -*/
+export class Shr implements Opcode {
+  static type: string = 'SHR';
+  static numberOfOperands = 3;
+
+  constructor(private aOffset: number, private bOffset: number, private destOffset: number) {}
+
+  execute(context: AvmContext, _stateManager: AvmStateManager): void {
+    const a: Fr = context.readMemory(this.aOffset);
+    const b: Fr = context.readMemory(this.bOffset);
+
+    // TODO: floor div? - this will not perform field division
+    const dest = new Fr(a.toBigInt() >> b.toBigInt());
+    context.writeMemory(this.destOffset, dest);
+  }
+}
```

Comment:
Do SHR and SHL operations belong in `bitwise.ts`?

Reply:
Good question, will move to arithmetic